User-friendly header links

peitalin opened the pull request:
Hi, 
I ran into some confusion trying to rebase and squash some merging commits (ended up creating a bunch of other unnecessary commits), so I deleted the other branch and tried again on a new branch.

Changes:
• replace header links with user-friendly slugs

jcheese1 commented:
@peitalin TY! ill review. you got a twitter or discord btw?

peitalin commented:
Yep, I'll dm you on discord

jcheese1 commented:
So we also dynamically get routes from graphql so we don't have to manually add a collection every time. what do you think of this:

Instead of having a `route` key in the hardcoded collection list, we derive it from the name. So for example, if the name is `Extra Life`, we call something like `name.replace(/\s+/g, '-').toLowerCase();` then we'll get `extra-life`

I think this way is a much more maintainable way since we can have arbitrary number of collections, plus it'll match the graphql side response as well. wdyt?

peitalin commented:
Sounds good to me, more maintainable.
I don't think we'll run into naming clashes between collections, so this should work.

peitalin commented:
Hey I've added the regex substitution with your suggestions, and it does look cleaner imo.

Also, swapped out some of the other links that used tokenAddress in the Listings component and inventory pages.

I've also renamed the `router.params.address` variable as `slugOrAddress` to reflect the change that it can either be a user-readable slug or token address (wasn't sure if you wanted to rename the `[address]/index.tsx` as `[slugOrAddress]/index.tsx` to change the query param name as the next.js level).

peitalin commented on src/pages/inventory/[[...section]].tsx:
```diff
@@ -661,139 +662,151 @@ const Inventory = () => {
+                                token.metadata?.image.includes("ipfs")
+                                  ? generateIpfsLink(token.metadata.image)
+                                  : token.metadata?.image ?? ""
                               }
```
No changes here, just an extra tab indent so I could add the slugOrAddress variable above

jcheese1 commented on src/lib/hooks.ts:
```diff
@@ -63,20 +63,28 @@ export function userFriendlyRouteToAddress(
   chainId: ChainId
 ) {
-  if (formattedAddress?.match(/^0x/)) {
+  if (formattedAddress?.match(/^(0[xX])[a-zA-Z0-9]{40}/)) {
```
or u can use the official `isAddress` helper from `ethers`

```
import { utils } from "ethers";

utils.isAddress(formattedAddress)
```

You can see an example of this in `src/pages/collection/[address]/[tokenId].tsx:1089`

jcheese1 commented on src/pages/collection/[address]/[tokenId].tsx:
```diff
@@ -85,7 +85,7 @@ export default function Example() {
   const queryClient = useQueryClient();
 
-  const { address, tokenId } = router.query;
+  const { address: slugOrAddress, tokenId } = router.query;
```
nice!

jcheese1 commented on src/lib/hooks.ts:
```diff
@@ -63,20 +63,28 @@ export function userFriendlyRouteToAddress(
 
+// takes a Collection Name and tries to return a user-friendly slug for routes
+// can return undefined if chainId is missing, or address lookup fails
+export function getCollectionSlugFromName(
```
maybe move this function and `userFriendlyRouteToAddress` to `src/utils.ts`?

peitalin commented on src/components/Listings.tsx:
```diff
@@ -141,6 +148,14 @@ const Listings = ({
               </thead>
               <tbody>
                 {listings.map((listing, index) => {
+                  const collectionName = getCollectionNameFromAddress(
```
Originally I was going to use `listing.collection.name` but decided to just look up the collection name with the existing `listing.collection.id` then converting it to a slug.

I tried adding `collection.name` to the `ListingFields` fragment here 
(https://github.com/TreasureProject/treasure-marketplace/blob/23308b104a0b665146272f1fd23612c6e3e38bf8/src/graphql/queries.graphql.ts#L184)
but it doesn't return any data (these queries are also generated by codegen so changes would be overwritten I'm guessing).

peitalin commented:
Ty for the suggestions; added those changes and made a few fixes on the `/activity` page (was using `token.name`, e.g. "Ancient Relic" instead of `collection.name` "Treasures".

jcheese1 commented:
did u close by mitake?

peitalin commented:
Yeah mistake, was trying to clean my main branch so I could fetch upstream commits more easily and keep up to date.
Let me make another PR with a cleaner commit history
